scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1052,7 +1052,36 @@ def cmake(self, filter_stages=[]):
         return self.run_cmake(args,filter_stages)
 
     def build(self):
-        return self.run_build(['--build', self.build_dir])
+        build_result = self.run_build(['--build', self.build_dir])
+        if self.testsuite.bsim_copy_exe and 'BSIM_OUT_PATH' in self.platform.env:
```

Comment:
nit: don't you think about logging warning here, if `bsim_copy_exe` is set and `BSIM_OUT_PATH` env var is missed?

Reply:
The condition `'BSIM_OUT_PATH' in self.platform.env` is used to verify if selected platform is a platform dedicated to work with BabbleSim (like `nrf52_bsim` or `nrf5340bsim`), because I would like to limit copying exe files only to this type of platforms. I can imagine, that there can exists test or sample, which is dedicated both for BabbleSim and non-BabbleSim platforms (e.g. `nrf52_bsim` and `nrf52840dk_nrf52840`) and it contains `bsim_copy_exe`. In this situation I would like to apply coping of exe file only for `nrf52_bsim` platform. So, in short words: I think that situation when `bsim_copy_exe` is set and `BSIM_OUT_PATH` does not exist in `self.platform.env` is valid and there shouldn't be any warning.
BTW verification if `self.platform.env` variables is missed is done by Twister itself here:
https://github.com/zephyrproject-rtos/zephyr/blob/f688532412432987682716de60bdf17f2afd4b38/scripts/pylib/twister/twisterlib/testplan.py#L776-L777